File: app/src/main/java/com/zuk0/gaijinsmash/riderz/di/module/ActivityModule.java
package com.zuk0.gaijinsmash.riderz.di.module;

import android.app.Activity;

import com.zuk0.gaijinsmash.riderz.ui.activity.main.MainActivity;
import com.zuk0.gaijinsmash.riderz.ui.activity.main.MainActivityComponent;
import com.zuk0.gaijinsmash.riderz.ui.activity.main.MainActivityModule;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.android.ContributesAndroidInjector;
import dagger.multibindings.IntoMap;

@Module
public abstract class ActivityModule {

    @ContributesAndroidInjector(modules = MainActivityModule.class)
    abstract MainActivity bindMainActivity();
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/di/module/ActivityModule.kt
package com.zuk0.gaijinsmash.riderz.di.module

import android.content.Context
import com.zuk0.gaijinsmash.riderz.ui.activity.main.fragment.home.presenter.HomeWeatherPresenter
import dagger.Module
import dagger.Provides

/**
 * place all activity scoped injections here
 */
@Module(includes = [FragmentBuilder::class])
class ActivityModule {

    @Provides
    fun provideHomeWeatherPresenter(context: Context) : HomeWeatherPresenter {
        return HomeWeatherPresenter(context)
    }
}